app/models.py

```diff
@@ -124,56 +139,90 @@ def create_from_images(images, project):
         return None
 
     def process(self):
-        # Nothing to do if we don't have a processing node...
-        if not self.processing_node: return
-        
-        # Need to process some images (UUID not yet set)?
-        if not self.uuid:
-            print("Processing... {}".format(self))
+        """
+        This method contains the logic for processing tasks asynchronously
+        from a background thread or from the scheduler. Here tasks that are
+        ready to be processed execute some logic. This could be communication
+        with a processing node or executing a pending action.
+        """
+
+        if self.processing_node:
+            # Need to process some images (UUID not yet set)?
+            if not self.uuid:
+                logger.info("Processing... {}".format(self))
+
+                images = [image.path() for image in self.imageupload_set.all()]
```

Comment:
`images` are paths to the local hard drive of the web server. They are saved by django in `app/media/project/<projectID>/task/<taskID>/filename.JPG`.

When a user uploads the images, they are stored there. Once all images have been uploaded, a processing node has been assigned and all options have been set, the images are sent to the processing node. The code you highlighted does this last step (sending the images to the processing node).